diff --git a/src/compiler/js-heap-broker.cc b/src/compiler/js-heap-broker.cc
index 2b81a9bd5ef..1dbeefe1e0d 100644
--- a/src/compiler/js-heap-broker.cc
+++ b/src/compiler/js-heap-broker.cc
@@ -23,6 +23,7 @@
 #include "src/objects/megadom-handler-inl.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/property-cell.h"
+#include "src/fuzzilli/cov.h"
 
 namespace v8 {
 namespace internal {
@@ -457,13 +458,15 @@ FeedbackSlotKind JSHeapBroker::GetFeedbackSlotKind(
     FeedbackSource const& source) const {
   if (HasFeedback(source)) return GetFeedback(source).slot_kind();
   FeedbackNexus nexus(source.vector, source.slot, feedback_nexus_config());
+  cov_add_feedback_nexus(&nexus);
   return nexus.kind();
 }
 
 bool JSHeapBroker::FeedbackIsInsufficient(FeedbackSource const& source) const {
   if (HasFeedback(source)) return GetFeedback(source).IsInsufficient();
-  return FeedbackNexus(source.vector, source.slot, feedback_nexus_config())
-      .IsUninitialized();
+  FeedbackNexus nexus = FeedbackNexus(source.vector, source.slot, feedback_nexus_config());
+  cov_add_feedback_nexus(&nexus);
+  return nexus.IsUninitialized();
 }
 
 const ProcessedFeedback& JSHeapBroker::NewInsufficientFeedback(
@@ -472,9 +475,11 @@ const ProcessedFeedback& JSHeapBroker::NewInsufficientFeedback(
 }
 
 ProcessedFeedback const& JSHeapBroker::ReadFeedbackForPropertyAccess(
-    FeedbackSource const& source, AccessMode mode, OptionalNameRef static_name,
+    FeedbackSource const& source, AccessMode mode,
+    OptionalNameRef static_name,
     bool allow_homomorphic) {
   FeedbackNexus nexus(source.vector, source.slot, feedback_nexus_config());
+  cov_add_feedback_nexus(&nexus);
   FeedbackSlotKind kind = nexus.kind();
   if (nexus.IsUninitialized()) return NewInsufficientFeedback(kind);
 
@@ -582,6 +587,7 @@ ProcessedFeedback const& JSHeapBroker::ReadFeedbackForPropertyAccess(
 ProcessedFeedback const& JSHeapBroker::ReadFeedbackForGlobalAccess(
     JSHeapBroker* broker, FeedbackSource const& source) {
   FeedbackNexus nexus(source.vector, source.slot, feedback_nexus_config());
+  cov_add_feedback_nexus(&nexus);
   DCHECK(nexus.kind() == FeedbackSlotKind::kLoadGlobalInsideTypeof ||
          nexus.kind() == FeedbackSlotKind::kLoadGlobalNotInsideTypeof ||
          nexus.kind() == FeedbackSlotKind::kStoreGlobalSloppy ||
@@ -626,6 +632,7 @@ ProcessedFeedback const& JSHeapBroker::ReadFeedbackForGlobalAccess(
 ProcessedFeedback const& JSHeapBroker::ReadFeedbackForBinaryOperation(
     FeedbackSource const& source) const {
   FeedbackNexus nexus(source.vector, source.slot, feedback_nexus_config());
+  cov_add_feedback_nexus(&nexus);
   if (nexus.IsUninitialized()) return NewInsufficientFeedback(nexus.kind());
   BinaryOperationHint hint = nexus.GetBinaryOperationFeedback();
   DCHECK_NE(hint, BinaryOperationHint::kNone);  // Not uninitialized.
@@ -635,6 +642,7 @@ ProcessedFeedback const& JSHeapBroker::ReadFeedbackForBinaryOperation(
 ProcessedFeedback const& JSHeapBroker::ReadFeedbackForTypeOf(
     FeedbackSource const& source) const {
   FeedbackNexus nexus(source.vector, source.slot, feedback_nexus_config());
+  cov_add_feedback_nexus(&nexus);
   if (nexus.IsUninitialized()) return NewInsufficientFeedback(nexus.kind());
   return *zone()->New<TypeOfOpFeedback>(nexus.GetTypeOfFeedback(),
                                         nexus.kind());
@@ -643,6 +651,7 @@ ProcessedFeedback const& JSHeapBroker::ReadFeedbackForTypeOf(
 ProcessedFeedback const& JSHeapBroker::ReadFeedbackForCompareOperation(
     FeedbackSource const& source) const {
   FeedbackNexus nexus(source.vector, source.slot, feedback_nexus_config());
+  cov_add_feedback_nexus(&nexus);
   if (nexus.IsUninitialized()) return NewInsufficientFeedback(nexus.kind());
   CompareOperationHint hint = nexus.GetCompareOperationFeedback();
   DCHECK_NE(hint, CompareOperationHint::kNone);  // Not uninitialized.
